tests/test_itd_service.rb

```diff
@@ -0,0 +1,291 @@
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/itd_device_group_node'
+require_relative '../lib/cisco_node_utils/itd_service'
+
+include Cisco
+# TestInterface - Minitest for general functionality
+# of the ItdService class.
+class TestItdService < CiscoTestCase
+  @skip_unless_supported = 'itd_service'
+  # Tests
+
+  def setup
+    super
+    config 'no feature itd'
+  end
+
+  def teardown
+    config 'no feature itd'
+    super
+  end
+
+  def test_itd_service_create_destroy
+    i1 = ItdService.new('abc')
+    i2 = ItdService.new('BCD')
+    i3 = ItdService.new('xyzABC')
+    assert_equal(3, ItdService.itds.keys.count)
+
+    i2.destroy
+    assert_equal(2, ItdService.itds.keys.count)
+
+    i1.destroy
+    i3.destroy
+    assert_equal(0, ItdService.itds.keys.count)
+  end
+
+  def test_access_list
+    itd = ItdService.new('new_group')
+    config 'ip access-list include'
+    config 'ip access-list exclude'
+    itd.access_list = 'include'
+    itd.exclude_access_list = 'exclude'
+    assert_equal('include', itd.access_list)
+    assert_equal('exclude', itd.exclude_access_list)
+    itd.access_list = itd.default_access_list
+    itd.exclude_access_list = itd.default_exclude_access_list
+    assert_equal(itd.default_access_list,
+                 itd.access_list)
+    assert_equal(itd.default_exclude_access_list,
+                 itd.exclude_access_list)
+    config 'no ip access-list include'
+    config 'no ip access-list exclude'
+    itd.destroy
```

Comment:
For `test_itd_service_create_destroy` you are validating the `destroy` method.  For this testcase though, you should not need an explicit destroy because you are issuing a `no feature itd` in the `teardown` method.